### ISSUE TITLE
Fatal Error using self::assertJsonContains in PhpUnit

### ISSUE THREAD

**Issue by rkeet** [api-platform/core#3726]
**API Platform version(s) affected**: Api Pack ^1.2 -> api-platform/core: v2.5.6

PhpUnit version: ^9.3

PHP: 7.4.8

**Description**  
Using `self::assertJsonContains` in PhpUnit results in Fatal Error: 

```
PHP Fatal error:  Declaration of ApiPlatform\Core\Bridge\Symfony\Bundle\Test\Constraint\ArraySubset::evaluate($other, string $description = '', bool $returnResult = false) must be compatible with PHPUnit\Framework\Constraint\Constraint::evaluate($o
ther, string $description = '', bool $returnResult = false): ?bool in E:\projects\workflow-poc\vendor\api-platform\core\src\Bridge\Symfony\Bundle\Test\Constraint\ArraySubset.php on line 46
```

**How to reproduce**  
Have a PhpUnit test extending `ApiTestCase` (`ApiPlatform\Core\Bridge\Symfony\Bundle\Test\ApiTestCase`). 

```
class SecretTest extends ApiTestCase
{
    use FixturesTrait; //Liip\TestFixturesBundle\Test\FixturesTrait

    private const URL = 'http://localhost/api/secrets';

    public function testSecretGetAll200Ok(): void
    {
        $client = static::createClient();
        
        $this->loadFixtures([SecretFixtures::class]);

        /** @var ResponseInterface $response */
        $response = $client->request('GET', self::URL );

        self::assertJsonContains( 
            [
                '@id' => '/api/secrets',
            ]
        );
```

**Possible Solution**  

Update function from (class: `ApiPlatform\Core\Bridge\Symfony\Bundle\Test\Constraint\ArraySubset`)
```
    public function evaluate($other, string $description = '', bool $returnResult = false):
```
To (add return type)
```
    public function evaluate($other, string $description = '', bool $returnResult = false): ?bool
```

**Comment by rkeet** [comment]
Seems related to [this](https://github.com/api-platform/core/issues/3153)

**Comment by Invis1bleReborn** [comment]
Try to set `SYMFONY_PHPUNIT_VERSION` env var, e.g. in `phpunit.xml`

```xml
<php>
    <env name="SYMFONY_PHPUNIT_VERSION" value="9.3" />
</php>
```

**Comment by dunglas** [comment]
Are you using the distribution or have you installed API Platform by yourself?

**Comment by rkeet** [comment]
@dunglas  Distribution I think, started the project using `composer require api` (as per [here](https://api-platform.com/docs/core/getting-started/#installing-api-platform-core)). 

@Invis1bleReborn Got this already. 

---

Just reran tests and they pass now. Checked the various `evaluate` functions used in the bridge, they all have a strict return type now. 

Checked the git blame and can only conclude I might've had [this version](https://github.com/api-platform/core/blame/ae350a9e1c28157608d94009555f96bb77c72629/src/Bridge/Symfony/Bundle/Test/Constraint/ArraySubset.php), though cannot fathom how I got one that old. 

Closing the issue as I might've had a mistake in version configuration.
